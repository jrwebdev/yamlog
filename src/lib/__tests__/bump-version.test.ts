import * as path from 'path';
import { copy, remove, move, readFile } from 'fs-extra';

import bumpVersion from '../bump-version';

let projectDir: string;
let projectBackupDir: string;

const setupTestProject = async (name: string) => {
  projectDir = path.resolve(__dirname, 'mock-projects', name);
  projectBackupDir = path.resolve(
    __dirname,
    'mock-projects',
    `.${name}-backup`
  );
  await copy(projectDir, projectBackupDir);
};

const restoreProjectDir = async () => {
  await remove(projectDir);
  await move(projectBackupDir, projectDir);
};

const readChangelog = async () => {
  const changelog = await readFile(path.resolve(projectDir, 'changelog.yaml'));
  return changelog.toString();
};

beforeEach(() => {
  expect.hasAssertions();
});

afterEach(() => {
  restoreProjectDir();
});

it('should bump a stable version', async () => {
  await setupTestProject('stable');
  await bumpVersion();
  const changelog = await readChangelog();
  expect(changelog).toMatchSnapshot();
});

it('should bump an unstable version', async () => {
  await setupTestProject('unstable');
  await bumpVersion({ unstable: true });
  const changelog = await readChangelog();
  expect(changelog).toMatchSnapshot();
});

it('should bump to version 1 on switching config from unstable to stable', async () => {
  await setupTestProject('unstable');
  await bumpVersion();
  const changelog = await readChangelog();
  expect(changelog).toMatchSnapshot();
});
