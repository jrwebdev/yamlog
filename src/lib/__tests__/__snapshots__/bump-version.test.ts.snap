// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should bump a stable version 1`] = `
|
  unreleased:
    breaking:
      - Breaking Change 1
    feature:
      - Feature 2
    fix:
      - Fix 2
  1.0.0:
    metadata:
      date: '2017-01-01'
    feature:
      - Feature 1
    fix:
      - Fix 1

`;
